src/System.Data.SqlClient/tests/ManualTests/SQL/ConnectionPoolTest/PoolBlockPeriodTest.netcoreapp.cs

```diff
@@ -0,0 +1,205 @@
+        }
+
+        public void PoolBlockingPeriodNonAzureTest(string connStr, PoolBlockingPeriod? policy)
+        {
+            SqlConnection.ClearAllPools();
+            Guid previousConnectionId = Guid.Empty;
+            int count = 0;
+
+            while (count < 2)
+            {
+                using (SqlConnection sqlConnection = new SqlConnection(connStr))
+                {
+                    try
+                    {
+                        sqlConnection.Open();
+                        throw new Exception("Connection Open must expect an exception!");
+                    }
+                    catch (SqlException e)
+                    {
+                        // if it is the first time the exception is happening (previousConnectionId == Guid.Empty) skip the check
+                        if (previousConnectionId != Guid.Empty)
+                        {
+                            switch (policy)
+                            {
+                                case PoolBlockingPeriod.Auto:
+                                case PoolBlockingPeriod.AlwaysBlock:
+                                    if (e.ClientConnectionId != previousConnectionId)
+                                    {
+                                        throw new Exception($"Connection Open with Policy '{policy}' expect an exception with same connection id!");
+                                    }
+                                    break;
+                                case PoolBlockingPeriod.NeverBlock:
+                                    if (e.ClientConnectionId == previousConnectionId)
+                                    {
+                                        throw new Exception($"Connection Open with Policy '{policy}' expect an exception with different connection id!");
+                                    }
+                                    break;
+                            }
+                        }
+                        previousConnectionId = e.ClientConnectionId;
+                    }
+                }
+                count++;
+            }
+        }
+
+        public void PoolBlockingPeriodAzureTest(string connStr, PoolBlockingPeriod? policy)
+        {
+            SqlConnection.ClearAllPools();
+            Guid previousConnectionId = Guid.Empty;
+            int count = 0;
+
+            while (count < 2)
+            {
+                using (SqlConnection sqlConnection = new SqlConnection(connStr))
+                {
+                    try
+                    {
+                        sqlConnection.Open();
+                        throw new Exception("Connection Open must expect an exception!");
+                    }
+                    catch (SqlException e)
+                    {
+                        // if it is the first time the exception is happening (previousConnectionId == Guid.Empty) skip the check
+                        if (previousConnectionId != Guid.Empty)
```

Comment:
Similar concern as above about entering this if condition

Here's my understanding:
When the SqlConnection is unable to establish a TCP connection, it sets an Empty Guid for the SqlConnection Connection Id. In your test you are setting previousConnectionId = e.ClientConnectionId which is effectively Guid.Empty.
Is your test entering this if condition?